python/ml4ir/base/pipeline.py

```diff
@@ -149,13 +149,6 @@ def validate_args(self):
                 )
             )
 
-        if self.optimizer_key not in OptimizerKey.get_all_keys():
```

Comment:
I removing this validation, because this is inherent to the `get_optimizer` function.